overlay.go

```diff
@@ -327,6 +327,16 @@ func (o *Overlay) TreeNodeFromToken(t *Token) (*TreeNode, error) {
 	return tn, nil
 }
 
+// Rx() implements the CounterIO interface
+func (c *Overlay) Rx() uint64 {
```

Comment:
This is a bit misleading since the overlay's bandwidth may not be the same as the server's bandwidth. Service's messages don't necessarily pass through the overlay. 
Since the `Overlay` is marshalling the message itself, it may be reasonable to implement accurately this interface counting only the messages it marshals and unmarshals ?
**(Same comment for TreeNodeInstance)**

Reply:
I'm looking at implementing this feature. If I'm understanding this correctly, for `Overlay`, we need to modify the `Wrap/Unwrap` functions from the `MessageProxy` interface to keep the tx/rx values. For `TreeNodeInstance`, we need to modify `dispatchMsgToProtocol/dispatchMsgReader` to do the same. Does that sound correct?